src/api/schema/components/sink.rs

```diff
@@ -5,25 +5,27 @@ use crate::{
         metrics::{self, IntoSinkMetrics},
         sort,
     },
+    config::ComponentId,
     filter_check,
 };
 use async_graphql::{Enum, InputObject, Object};
 use std::cmp;
 
 #[derive(Debug, Clone)]
 pub struct Data {
-    pub component_id: String,
+    pub component_id: ComponentId,
     pub component_type: String,
-    pub inputs: Vec<String>,
+    pub inputs: Vec<ComponentId>,
 }
 
 #[derive(Debug, Clone)]
 pub struct Sink(pub Data);
 
 impl Sink {
-    pub fn get_component_id(&self) -> &str {
-        self.0.component_id.as_str()
+    pub fn get_component_id(&self) -> &ComponentId {
```

Comment:
It's not obvious to me why `Component.get_component_id` returns a `String` (or a `&str` per my previous suggestion), yet a sink returns a `&ComponentId`.

Could these two be standardised?

Reply:
Agreed, I think I forgot the other one. 👍